app/Console/Commands/AdserversCrawlCommand.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Adshares\Adserver\Console\Commands;
+
+use Illuminate\Console\Command;
+
+use Adshares\Esc\Esc;
+
+use Adshares\Adserver\Models\NetworkCampaign;
+use Adshares\Adserver\Models\NetworkHost;
+use Adshares\Adserver\Services\Adselect;
+
+class AdserversCrawlCommand extends Command
+{
+    protected $broadcast=true;
+    protected $esc;
+    protected $host;
+    protected $registerHostsIfBroadcastedLimit = 3600; // seconds
+    protected $crawlHostsIfLastSeenLimit = 3600*24*14; // seconds
+
+
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'adshares:crawl';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Queries blockchain for available adsevers, downloads available advertisements from each adserver and stores offers in local db. Updates are forwarded to adselect.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->host = config('app.adserver_host');
+
+        // TODO: move to Service
+        // ($walletCommand, $workingDir, $address, $secret, $host, $port)
+        $this->esc = new Esc(
+            config('app.adshares_wallet'),
+            config('app.adshares_workdir'),
+            config('app.adshares_address'),
+            config('app.adshares_secret'),
+            config('app.adshares_node_host'),
+            config('app.adshares_node_port')
+        );
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        $this->readBroadcasts();
+        $this->crawlHosts();
+    }
+
+    protected function readBroadcasts()
+    {
+        try {
+            $logMessage = $this->esc->getBroadcastLog(time() - $this->registerHostsIfBroadcastedLimit);
+            print_r($logMessage);
+            $logs = $logMessage->broadcast;
+        } catch (\Exception $e) {
+            $logs = [
+                ['message' => bin2hex($this->host), 'address' => config('app.adshares_address'), 'account_msid' => 1],
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "address" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)